tests/test_spatial.py

```diff
@@ -766,115 +670,3 @@ def test_scales_lon_bounds_when_wrapping_around_prime_meridian(self):
-        expected = self.axis_weights["lat"]
-
-        assert result.identical(expected)
-
-    def test_weights_for_multiple_axis_is_the_product_of_matrix_multiplication(self):
-        result = self.ds.spatial._combine_weights(axis_weights=self.axis_weights)
-        expected = xr.DataArray(
-            data=np.array([[1, 2, 3, 4], [2, 4, 6, 8], [3, 6, 9, 12], [4, 8, 12, 16]]),
-            coords={"lat": self.ds.lat, "lon": self.ds.lon},
-            dims=["lat", "lon"],
-        )
-
-        assert result.identical(expected)
-
-
-class TestAverager:
-    @pytest.fixture(autouse=True)
-    def setup(self):
-        self.ds = generate_dataset(cf_compliant=True, has_bounds=True)
-
-    @requires_dask
-    def test_chunked_weighted_avg_over_lat_and_lon_axes(self):
-        ds = self.ds.copy().chunk(2)
-
-        weights = xr.DataArray(
-            data=np.array([[1, 2, 3, 4], [2, 4, 6, 8], [3, 6, 9, 12], [4, 8, 12, 16]]),
-            coords={"lat": ds.lat, "lon": ds.lon},
-            dims=["lat", "lon"],
-        )
-
-        result = ds.spatial._averager(ds.ts, axis=["X", "Y"], weights=weights)
-        expected = xr.DataArray(
-            name="ts", data=np.ones(15), coords={"time": ds.time}, dims=["time"]
-        )
-
-        assert result.identical(expected)
-
-    def test_weighted_avg_over_lat_axis(self):
-        weights = xr.DataArray(
-            name="lat_wts",
-            data=np.array([1, 2, 3, 4]),
-            coords={"lat": self.ds.lat},
-            dims=["lat"],
-        )
-
-        result = self.ds.spatial._averager(self.ds.ts, axis=["Y"], weights=weights)
-        expected = xr.DataArray(
-            name="ts",
-            data=np.ones((15, 4)),
-            coords={"time": self.ds.time, "lon": self.ds.lon},
-            dims=["time", "lon"],
-        )
-
-        assert result.identical(expected)
-
-    def test_weighted_avg_over_lon_axis(self):
-        weights = xr.DataArray(
-            name="lon_wts",
-            data=np.array([1, 2, 3, 4]),
-            coords={"lon": self.ds.lon},
-            dims=["lon"],
-        )
-
-        result = self.ds.spatial._averager(self.ds.ts, axis=["X"], weights=weights)
-        expected = xr.DataArray(
-            name="ts",
-            data=np.ones((15, 4)),
-            coords={"time": self.ds.time, "lat": self.ds.lat},
-            dims=["time", "lat"],
-        )
-
-        assert result.identical(expected)
-
-    def test_weighted_avg_over_lat_and_lon_axis(self):
-        weights = xr.DataArray(
-            data=np.array([[1, 2, 3, 4], [2, 4, 6, 8], [3, 6, 9, 12], [4, 8, 12, 16]]),
-            coords={"lat": self.ds.lat, "lon": self.ds.lon},
-            dims=["lat", "lon"],
-        )
-
-        result = self.ds.spatial._averager(self.ds.ts, axis=["X", "Y"], weights=weights)
-        expected = xr.DataArray(
-            name="ts", data=np.ones(15), coords={"time": self.ds.time}, dims=["time"]
-        )
-
-        assert result.identical(expected)
```

Comment:
I kind of remember you commenting on this...is this all removed because they are private methods (and we don't need to test them)?

Reply:
Same for `TestGetWeights`?

Reply:
Exactly, I consolidated tests by moving some of the tests for private methods to their parent public method.

No need to review the tests (aside my two comments for the failing tests) because we still have 100% test cover which means the tests for public methods cover the old tests for private methods.

Reply:
Correct, we should test the public method which will cover the private methods (implementation details). 

No need to review any of these changes here so I am just porting some tests over from the private methods to the public methods.